src/Reporting/src/App.Metrics.Reporting.Socket/SocketMetricsReporter.cs

```diff
@@ -61,26 +63,52 @@ public SocketMetricsReporter(MetricsReportingSocketOptions options)
         /// <inheritdoc />
         public async Task<bool> FlushAsync(MetricsDataValueSource metricsData, CancellationToken cancellationToken = default)
         {
-            Logger.Trace("Flushing metrics snapshot");
+            if (_socketClient.PreferChunkedOutput && Formatter is IMetricsChunkedOutputFormatter formatter)
+                return await WriteChunkedOutput(formatter, metricsData, cancellationToken);
+            return await WriteOutput(metricsData, cancellationToken);
+        }
+
+        private async Task<bool> WriteChunkedOutput(
+            IMetricsChunkedOutputFormatter formatter,
+            MetricsDataValueSource metricsData,
+            CancellationToken cancellationToken = default)
+        {
+            Logger.Trace("Flushing chunked metrics snapshot");
 
-            using (var stream = new MemoryStream())
+            var chunks = await formatter.WriteAsync(metricsData, cancellationToken);
+            foreach (var chunk in chunks)
             {
-                await Formatter.WriteAsync(stream, metricsData, cancellationToken);
+                var result = await _socketClient.WriteAsync(chunk, cancellationToken);
+                if (!result.Success)
+                {
+                    Logger.Error(result.ErrorMessage);
+                    return false;
```

Comment:
so what happens to the other chunks if one write fails?

Reply:
right now it bails out as soon as an error was caught, i guess we can try and send the rest of the data if one packet failed.

Reply:
Might as well, IMHO.